code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/message/views/Message.kt

```diff
@@ -62,13 +69,50 @@ internal fun MessageScreen(
         onBackPressed()
     }
 
-    Message(
-        isVisible = presentationStateManager.visibilityState,
-        inAppMessageSettings = inAppMessageSettings,
-        gestureTracker = gestureTracker,
-        onCreated = { onCreated(it) },
-        onDisposed = { onDisposed() }
-    )
+    if (inAppMessageSettings.shouldTakeOverUi) {
```

Comment:
I was looking into the broader code and realized that `Message` already has a clause that performs special rendering on `inAppMessageSettings.shouldTakeOverUi`. 

To avoid checks in multiple areas for the same condition (with a possibility of clashing or diverging later), I recommend that we move this conditional rendering into the `Message` composable. What are your thoughts on this?

```
@Composable
internal fun Message(
    isVisible: MutableTransitionState<Boolean>,
    inAppMessageSettings: InAppMessageSettings,
    gestureTracker: GestureTracker,
    onCreated: (WebView) -> Unit,
    onBackPressed: () -> Unit,
    onDisposed: () -> Unit
) {

    if (inAppMessageSettings.shouldTakeOverUi) {

        // Put content in a dialog
        Dialog(onDismissRequest = { onBackPressed() },
            properties = ....
        ) {
            // todo side effect handling for dim
           
            MessageBackdrop(
                visibility = isVisible,
                inAppMessageSettings = inAppMessageSettings,
                gestureTracker = gestureTracker
            )
            // Frame that holds the InAppMessage
            MessageFrame(
                visibility = isVisible,
                inAppMessageSettings = inAppMessageSettings,
                gestureTracker = gestureTracker,
                onCreated = onCreated,
                onDisposed = onDisposed
            )
        }

    } else {
        //just the frame that holds the InAppMessage
        MessageFrame(
            visibility = isVisible,
            inAppMessageSettings = inAppMessageSettings,
            gestureTracker = gestureTracker,
            onCreated = onCreated,
            onDisposed = onDisposed
        )
    }
}
```

Reply:
Done, I had the same thought at first. Thanks !